src/docs/message-handling.dox

```diff
@@ -0,0 +1,105 @@
+/*! @page message_handling Message handling
+
+@section message_handling_event_handler Message handling using the WT_EVENT_HANDLER
+
+Specific error and other message handling can be configured by passing an
+implementation of WT_EVENT_HANDLER to ::wiredtiger_open or
+WT_CONNECTION::open_session.
+
+For example, both informational and error messages might be passed to an
+application-specific logging function that added a timestamp and logged
+the message to a file, and error messages might additionally be output to
+the \c stderr file stream.
+
+Additionally, applications will normally handle \c WT_PANIC as a special
+case. WiredTiger will always call the error handler callback with
+\c WT_PANIC in the case of a fatal error requiring database restart,
+however, WiredTiger cannot guarantee applications will see an application
+thread return \c WT_PANIC from a WiredTiger API call. For this reason, a
+correctly-written WiredTiger application will likely specify at least an
```

Comment:
In what case does WiredTiger not call the error handler during a WT_PANIC? Also what would be the difference from a correctly written vs incorrectly written error handler?

Reply:
Same as my previous answer, this was existing content, however, this is my understanding:

> In what case does WiredTiger not call the error handler during a WT_PANIC?

As written above, I believe this is the opposite. When `WT_PANIC` occurs, the error handler will always be called.

> Also what would be the difference from a correctly written vs incorrectly written error handler?

When `WT_PANIC` error code is received, the application should stop as WireTiger should not keep going. The error handler should not let WiredTiger keep running after receiving `WT_PANIC` or it could lead to potential issues. As stated here, the minimum is to be able to handle fatal errors.